src/Spectre.Data/Datasets/BasicTextDataset.cs

```diff
@@ -0,0 +1,307 @@
+﻿/*
+ * BasicTextDataset.cs
+ * Class representing dataset created from streaming an ordinary text 
+ * file containing formatted data.
+ * 
+   Copyright 2017 Dariusz Kuchta
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using Spectre.Data.Structures;
+
+namespace Spectre.Data.Datasets
+{
+    public class BasicTextDataset : IDataset
+    {
+
+        //TODO: Because of not-so failsafe constructors, make the Datasets 
+        //TODO: be eventually spitted out by some factory class!!!
```

Comment:
I think, it would be helpful to keep such *TODOs* in one place - i.e. our Trello, not in code (at least when it comes to CR).